### ISSUE TITLE
most requirements skipped with conan 1.5 when some are private

### ISSUE THREAD

**Issue by drussel** [conan-io/conan#3137]
We have a conan setup that, when called with conan 1.4.5 or earlier, generates a visual studio property sheet with all our requirements. When called with 1.5, however, only one (Flow) of the requirements is included by the property sheet.

The conanfile.py that is used to generate the property sheet has a requirements field that looks like `requires = Flow.dependencies + ('Flow/'+Flow.Flow.version + "@velo3d/stable",)` where flow dependencies is a tuple like `dependencies = ('foo/1.2.3@velo3d/thirdparty',...` and the requires field on Flow looks like
```
    requires = dependencies + (
                ('X/b64300d87-5@velo3d/stable', 'private'),
                ('Y/d41fca23204-3@velo3d/stable', 'private'),
                ('Z/bcdf4b4f51a-2@velo3d/stable', 'private'),
                ('W/b155f94-1@velo3d/stable', 'private'),
                ('U/2d139f800c8-1@velo3d/stable', 'private'))
```

Going back to 1.4.5 or replacing Flow requires with
```
   requires = dependencies + (
                'X/b64300d87-5@velo3d/stable',
                'Y/d41fca23204-3@velo3d/stable',
                'Z/bcdf4b4f51a-2@velo3d/stable',
                'W/b155f94-1@velo3d/stable',
                'U/2d139f800c8-1@velo3d/stable')
```
fixes the problem.

Any ideas? Thanks.

**Comment by memsharded** [comment]
Hi @drussel !

Yes, this seems a regression. 

I am having a look, but I don't fully understand the relation between the consumer conanfile and the ``Flow``? Is that another conanfile? It is a requirement? If you could please add some more details, we will have a look asap and try to release a fix.

**Comment by drussel** [comment]
Sorry, yes.

We have a `Flow.py` defines the Flow package and a `conanfile.py` that defines dependencies for the consumer project. The latter reuses the dependencies tuple from `Flow.py` to define its `self.requires` (to which it adds Flow itself as a requirement). The Flow package also has some private requirements that it appends to the dependencies tuple when defining its `self.requires`.

Is that clearer?

**Comment by memsharded** [comment]
So, if I understood correctly, you want both the Flow package and the consumer conanfile.py package to have private requirements to all those packages. May I ask why those are private? It looks like you precisely want to make them public, so don't need the trick to use them from the dependency.

BTW, if you are in the CppLang slack team, please consider joining the #conan channel, it might be easier to discuss there.

**Comment by memsharded** [comment]
In any case, I will start running some tests, it is not clear how to reproduce yet, but I will start checking. Thanks for the feedback!

**Comment by memsharded** [comment]
Also, I am still missing how the consumer conanfile.py has access to ``Flow.dependencies``. The access to the ``Flow`` class is probably not a good idea, I wasn't aware myself that it was possible. How do you import it?

**Comment by drussel** [comment]
The intention was only for the flow package to have private dependencies (they are only added to the requires field for that package). The conanfile.py that is used to build the property sheet doesn't have them directly (but is does depend on the flow package and a bunch of others that flow also depends). Although somehow they end up as interpreted as private in 1.5.

Since the conanfile.py is a python file (and doesn't apparently need to be stand alone), it can import other python files (`import Flow`) and take values from variables defined in them. This might not be intended usage of Conan and there may be better ways to solve the problem of keeping dependency versions in sync without going nuts :-) Overall, having some way of managing versions of packages independently from the list of package names would be great.

I'll try to throw together a simple reproducing case when I'm in the office tomorrow. Thanks for your time. Conan really solves a need we have.

Cool, I'll check out the Conan slack channel.

**Comment by memsharded** [comment]
Really looking forward to getting your reproducible case, we will try to fix it as fast as possible.

> Overall, having some way of managing versions of packages independently from the list of package names would be great.

You mean, not having to specify versions in conanfiles? They would be global or something? Couldn't you use your current version ``self.version`` to define them?

See you in slack! Ask for the conan team there.

**Comment by drussel** [comment]
https://gist.github.com/drussel/7ef05d532cc71790a52bdd044bd15408 for MyPackage.py
https://gist.github.com/drussel/29fd8ff1364b9f2ca59d658a4f5ed403 for conanfile.py

command `python  -m conans.conan install --build=missing conanfile.py --update -s "compiler=Visual Studio" -s compiler.version=15 -s compiler.toolset=v140 -s build_type=Release -s compiler.runtime=MT -if .`

It appears to be fixed in 1.5.1.

**Comment by lasote** [comment]
1.5.2 Released! report if any issue.
